feed/utils.js

```diff
@@ -0,0 +1,68 @@
+async function getActivityFeedData(query = {}, nextLink) {
+  validateQuery(query);
+  let finalUrl =
+    API_BASE_URL + (nextLink || '/logs' + generateActivityFeedParams(query));
+  const res = await fetch(finalUrl, {
+    credentials: 'include',
+    method: 'GET',
+    headers: {
+      'Content-type': 'application/json',
+    },
+  });
+  return await res.json();
+}
+
+function generateActivityFeedParams(query) {
+  let queryParam = 'dev=true&format=feed';
+  if (query?.category) queryParam += getLogTypesFromCategory(query?.category);
+  return `?${queryParam}`;
+}
```

Comment:
the logic for constructing query parameters based on the category is repeated in multiple places. This logic can be abstracted into a reusable function to avoid duplication.

Reply:
I don't think that's it's repeating. Can you please highlight it where? also have made the generic function getLogTypesFromCategory() to construct the queryParams.

Reply:
`getLogTypesFromCategory` function contains a switch statement that handles different category types and returns corresponding log types. However, this logic is also repeated in the `generateActivityFeedParams` function. Extracting this logic into a separate function would eliminate duplication.

Reply:
@Achintya-Chatterjee  
generateActivityFeedParams is calling `getLogTypesFromCategory`, it's already extracted out.
<img width="793" alt="image" src="https://github.com/Real-Dev-Squad/website-dashboard/assets/61227144/429bb5b8-ae28-4813-a66f-7f37b15efac1">